### PR TITLE
created new branch AlgosOctWeek5

### PR DESCRIPTION
**Pull request by Serkebaeva**
new branch  algosOctWeek5

### CODE REVIEW

File: src/main/java/org/algorithms/ContainsDuplicate.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.algorithms;
+
+import java.util.Arrays;
+
+public class ContainsDuplicate {
+    public boolean containsDuplicate(int[] nums) {
+        Arrays.sort(nums);
```
Comment (asanaliev) [inline src/main/java/org/algorithms/ContainsDuplicate.java]: Can you do it without sorting?

File: src/main/java/org/algorithms/ContainsDuplicate.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        int temp = nums.length-1;
+
+        while(temp > 0) {
```
Comment (asanaliev) [inline src/main/java/org/algorithms/ContainsDuplicate.java]: Replace `while` with `for` not to use extra temp variable. Using an internal var inside `for` is ok.

File: src/main/java/org/algorithms/IndexOfFirstOccurrenceInAString.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package org.algorithms;
+
+public class IndexOfFirstOccurrenceInAString {
+    public int strStr(String haystack, String needle) {
+        for(int i = 0; i < haystack.length(); i++) {
+            return haystack.indexOf(needle);
```
Comment (asanaliev) [inline src/main/java/org/algorithms/IndexOfFirstOccurrenceInAString.java]: Don't use ready method. Implement it yourself. Assume there is no `indexOf` ready.

File: src/main/java/org/algorithms/PalindromeNumber.java
Comment (asanaliev) [inline src/main/java/org/algorithms/PalindromeNumber.java:1]: > Could you solve it without converting the integer to a string?

**Review by asanaliev — CHANGES_REQUESTED** [review]
Also, please read Google Java Style Guide and apply their rules in your code.

File: src/main/java/org/algorithms/ContainsDuplicate.java (hunk 1 above)
Comment (Serkebaeva) [reply src/main/java/org/algorithms/ContainsDuplicate.java]: @asanaliev  Contains duplicate: changed to For loop, without sorting, and Google Java Style Guide changed indentation to 2 in JVM, classes UpperCamelCased, methods lowerCamelCased, added vertical whitespaces for readability sake, am I missing anything here?

File: src/main/java/org/algorithms/ContainsDuplicate.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package org.algorithms;
+
+public class ContainsDuplicate {
+
+  public boolean containsDuplicate(int[] nums) {
+    for (int i = 0; i < nums.length; i++) {
```
Comment (asanaliev) [inline src/main/java/org/algorithms/ContainsDuplicate.java]: Works but using two loops is very inefficient. Try to solve using Set or Map, and one loop only.

**Review by asanaliev — CHANGES_REQUESTED** [review]

File: src/main/java/org/algorithms/ContainsDuplicate.java (hunk 4 above)
Comment (Serkebaeva) [reply src/main/java/org/algorithms/ContainsDuplicate.java]: @asanaliev done

**Review by asanaliev — APPROVED** [review]
Let's merge the PR.

**Pull request merged**